### PR TITLE
Fix when duplicating product, out of stock ordering behavior is not copied

### LINKED ISSUE #25448: When duplicating product, out of stock ordering behavior is not copied
#### Describe the bug

When duplicating product, out of stock ordering behavior is not copied

#### Expected behavior

out of stock ordering behavior is copied too

#### Steps to Reproduce

1. Create a product, set quantity to zero, disable ordering when out of stock.
2. Duplicate product, open it.
3. See that ordering out of stock got reset to "Default".

#### Additional information

* PrestaShop version: 1.7.8 beta
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by gennaris**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Fixes ps_stock_available "out_of_stock" status not being copied when duplicating a product
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     |no
| Fixed ticket?     | Fixes #25448 
| How to test?      | Follow 25448 reproduction steps before / after this patch
| Possible impacts? | N/A

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25787)

### CODE REVIEW

**Comment by matks** [comment]
Close and reset for CI

**Review by Progi1984 — COMMENTED** [review]
Could you add a test for this behavior change ?

**Comment by PierreRambaud** [comment]
> Could you add a test for this behavior change ?

I don't think the current implementation is testable :/

**Review by matks — DISMISSED** [review]
I agree with @PierreRambaud . I approve the PR.

File: src/Adapter/Product/AdminProductDataUpdater.php [hunk 1]
```diff
@@ -233,7 +234,8 @@ public function duplicateProduct($productId, $namePattern = 'copy of %s')
                 if (in_array($product->visibility, ['both', 'search']) && Configuration::get('PS_SEARCH_INDEXATION')) {
                     Search::indexation(false, $product->id);
                 }
-
+                StockAvailable::setProductOutOfStock($product->id, StockAvailable::outOfStock($id_product_old));
+
```
Comment (Progi1984) [inline src/Adapter/Product/AdminProductDataUpdater.php]:
```suggestion

```

Fixes PHPCSFixer errors.

**Review by Progi1984 — COMMENTED** [review]
@gennaris Thank you for your contribution. I have brought you a small comment for fixing PHPCSFixer errors.

File: src/Adapter/Product/AdminProductDataUpdater.php [hunk 2]
```diff
@@ -233,7 +234,7 @@ public function duplicateProduct($productId, $namePattern = 'copy of %s')
                 if (in_array($product->visibility, ['both', 'search']) && Configuration::get('PS_SEARCH_INDEXATION')) {
                     Search::indexation(false, $product->id);
                 }
-
+                StockAvailable::setProductOutOfStock($product->id, StockAvailable::outOfStock($id_product_old));
                 return $product->id;
```
Comment (Progi1984) [inline src/Adapter/Product/AdminProductDataUpdater.php:239]:
```suggestion

                return $product->id;
```

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by florine2623** [comment]
Hello @gennaris ,

Thanks for your PR !
Tested and approved !

**Comment by Progi1984** [comment]
Thanks @gennaris & @florine2623

**Pull request merged**